rockcraft/application.py

```diff
@@ -0,0 +1,111 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Main Rockcraft Application."""
+
+from __future__ import annotations
+
+import os
+import signal
+import sys
+
+import craft_cli
+from craft_application import Application, AppMetadata
+from craft_application.application import _Dispatcher
+from overrides import override
```

Comment:
Isn't typing_extensions also providing override? That could be preferable if functionality is equivalent.